agents/postgres/pgstatmonitor/stat_monitor_cache.go

```diff
@@ -150,16 +150,30 @@ func (ssc *statMonitorCache) getStatMonitorExtended(ctx context.Context, q *refo
 			example := ""
 			if !normalizedQuery {
 				example = c.Query
-				fingerprint = ssc.generateFingerprint(c.Query)
+				fingerprint, err = ssc.generateFingerprint(c.Query)
 			}
-			var isTruncated bool
-			c.Fingerprint, isTruncated = truncate.Query(fingerprint)
-			if isTruncated {
-				c.IsQueryTruncated = isTruncated
-			}
-			c.Example, isTruncated = truncate.Query(example)
-			if isTruncated {
-				c.IsQueryTruncated = isTruncated
+			if err != nil {
+				// Either real syntax error in the query or pg_stat_monitor truncated the query and it causes the syntax error.
+				if c.pgStatMonitor.Elevel != 0 {
+					c.IsQueryTruncated = false
+					ssc.l.Warnf("generating fingerprint failed for query with id %v: %v", c.QueryID, err)
+				} else {
+					// We failed to normalize the query but the query was executed successfully.
+					// That means the query was truncated by pg_stat_monitor. Let's state the query was truncated.
+					c.IsQueryTruncated = true
+				}
+				c.Example = c.Query
+				c.Fingerprint = c.Query
```

Comment:
we should discuss it with @rnovikovP, since we show query as is which might be not secure.

Reply:
I'm ok with queries in logs when it's an error for them. 
We'll have the log level controlled from the PMM side, so for some secured env. this will be possible to OFF